### DISCUSSION TITLE
安装alist+emby，打开Ip:2345，出现502 badgateway (nginx)，怎么解决？

### DISCUSSION THREAD

**Discussion by ywqboy3501** [monlor/docker-xiaoya#168] [General]
安装alist+emby，打开Ip:2345，出现502 badgateway (nginx)，怎么解决？

**Comment by monlor** [comment]
看日志，不是安装好就立马能用

**Reply by gh279760559** [reply]
你好,我也遇见了同样的问题,logs里没有看到错误, 前面一长串的下载我就不贴出来了,贴出来的是最后一段.
无论是按照提示说的ip+2345(502 bad gateway) 还是8096 8920 都无法访问.
使用emby 客户端, host填入地址, 端口填入上述的端口都无法连接
```
metadata-1  | 07/10 20:35:30 [NOTICE] Download complete: /media/temp/pikpak.mp4
metadata-1  |
metadata-1  | Download Results:
metadata-1  | gid   |stat|avg speed  |path/URI
metadata-1  | ======+====+===========+=======================================================
metadata-1  | 57a8f5|OK  |    47MiB/s|/media/temp/pikpak.mp4
metadata-1  |
metadata-1  | Status Legend:
metadata-1  | (OK):download completed.
metadata-1  | Extracting Emby media...
metadata-1  |
metadata-1  | 7-Zip [64] 16.02 : Copyright (c) 1999-2016 Igor Pavlov : 2016-05-21
metadata-1  | p7zip Version 16.02 (locale=zh_CN.UTF-8,Utf16=on,HugeFiles=on,64 bits,12 CPUs Intel(R) Core(TM) i5-10500 CPU @ 3.10GHz (A0653),ASM,AES-NI)
metadata-1  |
metadata-1  | Scanning the drive for archives:
metadata-1  | 1 file, 35999219160 bytes (34 GiB)
metadata-1  |
metadata-1  | Extracting archive: /media/temp/all.mp4
metadata-1  | 检查alist连通性...
metadata-1  | alist启动完成，等待10秒后开始下载元数据...
metadata-1  | 开始下载元数据，如果有问题无法解决，请删除目录 /media/temp 下的所有文件重新启动.
metadata-1  | 目录/media/temp最少需要剩余空间：5G，目前仅剩：1G
metadata-1  | 检查alist连通性...
metadata-1  | alist启动完成，等待10秒后开始下载元数据...
metadata-1  | 开始下载元数据，如果有问题无法解决，请删除目录 /media/temp 下的所有文件重新启动.
metadata-1  | Emby metadata has been downloaded. Delete the file /media/config/emby_meta_finished to re-extract.
metadata-1  | Cleaning up Emby media...
metadata-1  | Downloading Emby media...
metadata-1  | Downloading all.mp4...
metadata-1  | Found all.mp4, skip.
metadata-1  | Downloading pikpak.mp4...
metadata-1  | Found pikpak.mp4, skip.
metadata-1  | Extracting Emby media...
metadata-1  |
metadata-1  | 7-Zip [64] 16.02 : Copyright (c) 1999-2016 Igor Pavlov : 2016-05-21
metadata-1  | p7zip Version 16.02 (locale=zh_CN.UTF-8,Utf16=on,HugeFiles=on,64 bits,12 CPUs Intel(R) Core(TM) i5-10500 CPU @ 3.10GHz (A0653),ASM,AES-NI)
metadata-1  |
metadata-1  | Scanning the drive for archives:
metadata-1  | 1 file, 35999219160 bytes (34 GiB)
metadata-1  |
metadata-1  | Extracting archive: /media/temp/all.mp4
metadata-1  | --
metadata-1  | Path = /media/temp/all.mp4
metadata-1  | Type = 7z
metadata-1  | Offset = 370745
metadata-1  | Physical Size = 35998848415
metadata-1  | Headers Size = 8897151
metadata-1  | Method = LZMA2:24
metadata-1  | Solid = +
metadata-1  | Blocks = 22
metadata-1  |
metadata-1  | Everything is Ok
metadata-1  |
metadata-1  | Folders: 34363
metadata-1  | Files: 864706
metadata-1  | Size:       45748897719
metadata-1  | Compressed: 35999219160
metadata-1  |
metadata-1  | 7-Zip [64] 16.02 : Copyright (c) 1999-2016 Igor Pavlov : 2016-05-21
metadata-1  | p7zip Version 16.02 (locale=zh_CN.UTF-8,Utf16=on,HugeFiles=on,64 bits,12 CPUs Intel(R) Core(TM) i5-10500 CPU @ 3.10GHz (A0653),ASM,AES-NI)
metadata-1  |
metadata-1  | Scanning the drive for archives:
metadata-1  | 1 file, 15168649913 bytes (15 GiB)
metadata-1  |
metadata-1  | Extracting archive: /media/temp/pikpak.mp4
metadata-1  | --
metadata-1  | Path = /media/temp/pikpak.mp4
metadata-1  | Type = 7z
metadata-1  | Offset = 370745
metadata-1  | Physical Size = 15168279168
metadata-1  | Headers Size = 2974843
metadata-1  | Method = LZMA2:26
metadata-1  | Solid = +
metadata-1  | Blocks = 5
metadata-1  |
metadata-1  | Everything is Ok
metadata-1  |
metadata-1  | Folders: 13456
metadata-1  | Files: 312730
metadata-1  | Size:       20640872458
metadata-1  | Compressed: 15168649913
metadata-1  | 启动定时更新Emby配置任务...
metadata-1  | 启动定时更新Emby媒体数据任务...
metadata-1  | Complete.
```

**Reply by monlor** [reply]
> 你好,我也遇见了同样的问题,logs里没有看到错误, 前面一长串的下载我就不贴出来了,贴出来的是最后一段.
> 
> 无论是按照提示说的ip+2345(502 bad gateway) 还是8096 8920 都无法访问.
> 
> 使用emby 客户端, host填入地址, 端口填入上述的端口都无法连接
> 
> ```
> 
> metadata-1  | 07/10 20:35:30 [NOTICE] Download complete: /media/temp/pikpak.mp4
> 
> metadata-1  |
> 
> metadata-1  | Download Results:
> 
> metadata-1  | gid   |stat|avg speed  |path/URI
> 
> metadata-1  | ======+====+===========+=======================================================
> 
> metadata-1  | 57a8f5|OK  |    47MiB/s|/media/temp/pikpak.mp4
> 
> metadata-1  |
> 
> metadata-1  | Status Legend:
> 
> metadata-1  | (OK):download completed.
> 
> metadata-1  | Extracting Emby media...
> 
> metadata-1  |
> 
> metadata-1  | 7-Zip [64] 16.02 : Copyright (c) 1999-2016 Igor Pavlov : 2016-05-21
> 
> metadata-1  | p7zip Version 16.02 (locale=zh_CN.UTF-8,Utf16=on,HugeFiles=on,64 bits,12 CPUs Intel(R) Core(TM) i5-10500 CPU @ 3.10GHz (A0653),ASM,AES-NI)
> 
> metadata-1  |
> 
> metadata-1  | Scanning the drive for archives:
> 
> metadata-1  | 1 file, 35999219160 bytes (34 GiB)
> 
> metadata-1  |
> 
> metadata-1  | Extracting archive: /media/temp/all.mp4
> 
> metadata-1  | 检查alist连通性...
> 
> metadata-1  | alist启动完成，等待10秒后开始下载元数据...
> 
> metadata-1  | 开始下载元数据，如果有问题无法解决，请删除目录 /media/temp 下的所有文件重新启动.
> 
> metadata-1  | 目录/media/temp最少需要剩余空间：5G，目前仅剩：1G
> 
> metadata-1  | 检查alist连通性...
> 
> metadata-1  | alist启动完成，等待10秒后开始下载元数据...
> 
> metadata-1  | 开始下载元数据，如果有问题无法解决，请删除目录 /media/temp 下的所有文件重新启动.
> 
> metadata-1  | Emby metadata has been downloaded. Delete the file /media/config/emby_meta_finished to re-extract.
> 
> metadata-1  | Cleaning up Emby media...
> 
> metadata-1  | Downloading Emby media...
> 
> metadata-1  | Downloading all.mp4...
> 
> metadata-1  | Found all.mp4, skip.
> 
> metadata-1  | Downloading pikpak.mp4...
> 
> metadata-1  | Found pikpak.mp4, skip.
> 
> metadata-1  | Extracting Emby media...
> 
> metadata-1  |
> 
> metadata-1  | 7-Zip [64] 16.02 : Copyright (c) 1999-2016 Igor Pavlov : 2016-05-21
> 
> metadata-1  | p7zip Version 16.02 (locale=zh_CN.UTF-8,Utf16=on,HugeFiles=on,64 bits,12 CPUs Intel(R) Core(TM) i5-10500 CPU @ 3.10GHz (A0653),ASM,AES-NI)
> 
> metadata-1  |
> 
> metadata-1  | Scanning the drive for archives:
> 
> metadata-1  | 1 file, 35999219160 bytes (34 GiB)
> 
> metadata-1  |
> 
> metadata-1  | Extracting archive: /media/temp/all.mp4
> 
> metadata-1  | --
> 
> metadata-1  | Path = /media/temp/all.mp4
> 
> metadata-1  | Type = 7z
> 
> metadata-1  | Offset = 370745
> 
> metadata-1  | Physical Size = 35998848415
> 
> metadata-1  | Headers Size = 8897151
> 
> metadata-1  | Method = LZMA2:24
> 
> metadata-1  | Solid = +
> 
> metadata-1  | Blocks = 22
> 
> metadata-1  |
> 
> metadata-1  | Everything is Ok
> 
> metadata-1  |
> 
> metadata-1  | Folders: 34363
> 
> metadata-1  | Files: 864706
> 
> metadata-1  | Size:       45748897719
> 
> metadata-1  | Compressed: 35999219160
> 
> metadata-1  |
> 
> metadata-1  | 7-Zip [64] 16.02 : Copyright (c) 1999-2016 Igor Pavlov : 2016-05-21
> 
> metadata-1  | p7zip Version 16.02 (locale=zh_CN.UTF-8,Utf16=on,HugeFiles=on,64 bits,12 CPUs Intel(R) Core(TM) i5-10500 CPU @ 3.10GHz (A0653),ASM,AES-NI)
> 
> metadata-1  |
> 
> metadata-1  | Scanning the drive for archives:
> 
> metadata-1  | 1 file, 15168649913 bytes (15 GiB)
> 
> metadata-1  |
> 
> metadata-1  | Extracting archive: /media/temp/pikpak.mp4
> 
> metadata-1  | --
> 
> metadata-1  | Path = /media/temp/pikpak.mp4
> 
> metadata-1  | Type = 7z
> 
> metadata-1  | Offset = 370745
> 
> metadata-1  | Physical Size = 15168279168
> 
> metadata-1  | Headers Size = 2974843
> 
> metadata-1  | Method = LZMA2:26
> 
> metadata-1  | Solid = +
> 
> metadata-1  | Blocks = 5
> 
> metadata-1  |
> 
> metadata-1  | Everything is Ok
> 
> metadata-1  |
> 
> metadata-1  | Folders: 13456
> 
> metadata-1  | Files: 312730
> 
> metadata-1  | Size:       20640872458
> 
> metadata-1  | Compressed: 15168649913
> 
> metadata-1  | 启动定时更新Emby配置任务...
> 
> metadata-1  | 启动定时更新Emby媒体数据任务...
> 
> metadata-1  | Complete.
> 
> ```

没有贴emby的日志 emby应该没启动完

**Reply by ltm0203** [reply]
想请问下 这个数据配置的时间大概是多久：

![QQ_1722364991576](https://github.com/user-attachments/assets/9a01697e-b2cc-463d-88f1-69b99b43c3fc)

我看到源代码有这个检测；


可能我的是虚拟机，一直是以下的info信息，时间蛮久了。
```bash
root@vodnasserver:/home/docker/nas_xiaoya# docker logs -f nas_xiaoya-emby-1
检查alist连通性...
等待配置数据下载完成...
```

```bash

alist-1     | 07-31-2024 02:42:33 Alist_V3 已成功缓存了 1 个文件
metadata-1  | Downloading config.mp4...
metadata-1  | Found config.mp4, skip.
metadata-1  | Extracting Emby config...
metadata-1  | 
metadata-1  | 7-Zip [64] 16.02 : Copyright (c) 1999-2016 Igor Pavlov : 2016-05-21
metadata-1  | p7zip Version 16.02 (locale=C.UTF-8,Utf16=on,HugeFiles=on,64 bits,8 CPUs Intel(R) Xeon(R) CPU E5-2699 v3 @ 2.30GHz (306F2),ASM,AES-NI)
metadata-1  | 
metadata-1  | Scanning the drive for archives:
metadata-1  | 1 file, 4665436082 bytes (4450 MiB)
metadata-1  | 
metadata-1  | Extracting archive: temp/config.mp4

```

**Reply by ltm0203** [reply]
![QQ_1722365520627](https://github.com/user-attachments/assets/3ff374eb-e0e9-4525-8b00-ba2c191004ca)

**Reply by monlor** [reply]
没会员下一天也很正常🙃

**Reply by ltm0203** [reply]
> 没会员下一天也很正常🙃

嗯，最近因为有时间所以折腾了下NAS，发现小雅这么一个项目，你的仓库帮我节省了太多时间了。非常感谢；但是发现就是我没有赶上好时候，恰好这段时间是阿里云限速的时候，必须开阿里云盘会员；不然想在线看4K就是痴人说梦了。哈哈哈；不过对付对付看1080P的动漫还是可以的 ；

**Reply by AvenCoder** [reply]
我开了会员，显示元数据下载完成，但是emby还是卡等待
![image](https://github.com/user-attachments/assets/ffcd3580-85ce-4362-bd57-782621a695ad)

**Reply by monlor** [reply]
> 我开了会员，显示元数据下载完成，但是emby还是卡等待
> 
> ![image](https://github.com/user-attachments/assets/ffcd3580-85ce-4362-bd57-782621a695ad)
> 
> 

emby服务有自动重启么
